connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -553,12 +547,8 @@ public void testErrorInRebalancePartitionLoss() {
         workerTask.iteration();
         verifyPollInitialAssignment();
 
-        try {
-            workerTask.iteration();
-            fail("Poll should have raised the rebalance exception");
-        } catch (RuntimeException e) {
-            assertEquals(exception, e);
-        }
+        Throwable thrownException = assertThrows(RuntimeException.class, () -> workerTask.iteration());
```

Comment:
The type of the `thrownException` should be `RuntimeException` as it has been casted when we called `assertThrows(RuntimeException.class, <lambda>)`